backend/src/routes/weaviate.ts

```diff
@@ -45,15 +54,61 @@ weaviateRouter.get("/", async (_req, res) => {
 weaviateRouter.get("/test", async (_req, res) => {
   const result = await client.graphql
     .get()
-    .withClassName('ResearchPaper')
+    .withClassName("ResearchPaper")
     .withHybrid({
-      query: 'software',
+      query: "software",
     })
     .withLimit(3)
-    .withFields('title content')
+    .withFields("title content")
     .do();
 
   res.status(201).json({ response: JSON.stringify(result, null, 2) });
 });
 
+weaviateRouter.post("/search", async (req, res) => {
+  const inputString = req.body.text;
+  console.log(inputString);
+
+  if (!inputString) {
+    res.status(400).json({ error: "text is required in request body" });
+  }
+
+  const promptText = `You are an entity extractor now. Extract entities from the following text: '${inputString}'`;
+  try {
+    const response = await axios.post(
+      OPEN_AI_API_URL,
+      {
+        model: "gpt-3.5-turbo",
+        messages: [{ role: "system", content: promptText }],
+        temperature: 0.7,
```

Comment:
Food for thought but lets adjust it later.

https://gptforwork.com/guides/openai-gpt3-temperature

"For transformation tasks (extraction, standardization, format conversion, grammar fixes) prefer a temperature of 0 or up to 0.3.
For writing tasks, you should juice the temperature higher, closer to 0.5. If you want GPT to be highly creative (for marketing or advertising copy for instance), consider values between 0.7 and 1."